library/src/main/scala/scala/scalajs/reflect/Reflect.scala

```diff
@@ -0,0 +1,95 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js API               **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2017, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package scala.scalajs.reflect
+
+import scala.collection.mutable
+
+import scala.scalajs.js
+
+final class LoadableModuleClass private[reflect] (
+    val runtimeClass: Class[_],
+    loadModuleFun: js.Function0[Any]
+) {
+  /** Loads the module instance and returns it. */
+  def loadModule(): Any = loadModuleFun()
+}
+
+final class InstantiatableClass private[reflect] (
+    val runtimeClass: Class[_],
+    val declaredConstructors: List[InvokableConstructor]
+) {
+  /** Instantiates a new instance of this class using the zero-argument
+   *  constructor.
+   *
+   *  @throws java.lang.InstantiationException (caused by a
+   *    `NoSuchMethodException`)
+   *    If this class is abstract or does not have a public zero-argument
```

Comment:
Indeed. The comment is obsolete. At some point I did return an `InstantiatableClass` with no `declaredConstructors`, but then I thought better of it (for the reason that you give).